src/index.js

```diff
@@ -0,0 +1,276 @@
+    profileDescription.textContent = userInfo.about;
+    const profileAvatar = document.querySelector(".profile__avatar");
+    profileAvatar.src = userInfo.avatar;
+
+    cards.forEach((item) => {
+      const cardEl = getCardElement(item);
+      cardsList.append(cardEl);
+    });
+  })
+  .catch(console.error);
+
+function openModal(modal) {
+  modal.classList.add("modal_opened");
+  addEscapeListener(); // Add
+}
+
+function closeModal(modal) {
+  modal.classList.remove("modal_opened");
+  removeEscapeListener(); // Remove
+}
+
+function handleEditFormSubmit(evt) {
+  evt.preventDefault();
+  editSubmitButton.textContent = "Saving...";
+  api
+    .editUserInfo({
+      name: editModalNameInput.value,
+      about: editModalDescriptionInput.value,
+    })
+    .then((data) => {
+      profileName.textContent = data.name;
+      profileDescription.textContent = data.about;
+      closeModal(editModal);
+      editSubmitButton.textContent = "Save";
+    })
+    .catch(console.error);
+}
+
+function handleAddCardSubmit(evt) {
+  evt.preventDefault();
+  const inputValues = { name: cardNameInput.value, link: cardLinkInput.value };
+  cardSubmitButton.textContent = "Saving...";
+  api
+    .addNewCard(inputValues)
+    .then((card) => {
+      const cardElement = getCardElement(card);
+      cardsList.prepend(cardElement);
+      closeModal(cardModal);
+      cardForm.reset();
+      disableButton(evt.submitter, settings);
+      cardSubmitButton.textContent = "Save";
+    })
+    .catch(console.error);
+}
+
+changeAvatarForm.addEventListener("submit", (event) => {
+  event.preventDefault();
+
+  const formData = new FormData(event.target);
+  avatarSubmitButton.textContent = "Saving...";
+  // output as an object
+  const data = Object.fromEntries(formData);
+  api
+    .editAvatar(data)
+    .then(() => {
+      const avatar = document.querySelector(".profile__avatar");
+      avatar.src = data.avatar;
+      closeModal(changeAvatarModal);
+      avatarSubmitButton.textContent = "Save";
+    })
+    .catch(console.error);
+});
+
+let deleteListener = null;
+function getCardElement(data) {
+  const cardElement = cardTemplate.content
+    .querySelector(".card")
+    .cloneNode(true);
+
+  const cardNameEl = cardElement.querySelector(".card__title");
+  const cardImgEl = cardElement.querySelector(".card__image");
+  const cardLikeBtn = cardElement.querySelector(".card__like-btn");
+  const cardDeleteBtn = cardElement.querySelector(".card__delete-btn");
+
+  cardImgEl.src = data.link;
+  cardImgEl.alt = data.name;
+  cardNameEl.textContent = data.name;
+
+  if (data.isLiked) {
+    cardLikeBtn.classList.add("card__like-btn_liked");
+  }
+  cardLikeBtn.addEventListener("click", () => {
+    if (cardLikeBtn.classList.contains("card__like-btn_liked")) {
+      api.removeLike(data._id).catch(console.error);
+    } else {
+      api.addLike(data._id).catch(console.error);
+    }
+    cardLikeBtn.classList.toggle("card__like-btn_liked");
```

Comment:
You need to **toggle likes** only in a `then` block after a successful response from the server because if there is an error, the user will not know about that and will see the toggled heart color.